### PR TITLE
kafka: FIXES images/updates SendNotifications.

### PR DESCRIPTION
**Pull request by ldjebran**
# Description

1. Images SendNotification call Flush before closing the producer. (Using flush scenario was not working as was blocking all the application for that amount of time , and was replaced with WaitProduceDeliveryEvent that is working )
2. Update SendNotification use the project GetProducerInstance that is a global producer, use deliveryChannel to ensure, the message is delivered or not. see https://github.com/confluentinc/confluent-kafka-go/blob/80c58f81b6cc32d3ed046609bf660a41a061b23d/examples/producer_example/producer_example.go

FIXES-THEEDGE-3000

## Type of change

What is it?

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [X] Tests update
- [ ] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

File: pkg/common/kafka/utils.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	"errors"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+)
+
+var ErrUnknownProduceDeliveryEvent = errors.New("producer produce returned unknown delivery event message")
+
+// WaitProduceDeliveryEvent wait for a message to be delivered used with Producer Produce function
+// this allows to not block other deliveries when using a global producer instance (ProducerService.GetProducerInstance())
+func WaitProduceDeliveryEvent(deliveryChan chan kafka.Event) error {
```
[the remark this answers is not in the document]
Comment (ldjebran) [reply pkg/common/kafka/utils.go]: added a timeout functionality and a unittest timeout case :smiley:

File: pkg/services/images.go [hunk 2]
```diff
@@ -1610,17 +1610,31 @@ func (s *ImageService) SendImageNotification(i *models.Image) (ImageNotification
...
-			s.log.WithField("message", perr.Error()).Error("Error on produce")
+			s.log.WithFields(log.Fields{"message": perr.Error(), "topic": topic}).Error("Error on produce")
+			return notify, perr
+		}
+		// flush message before closing Producer with a timeout of 10 sec
+		// TODO uncomment
+		// p.Flush(kafkacommon.DefaultDeliveryTimeout)
+		// wait for notification delivery
+		err = WaitNotificationDelivery(deliveryChan, kafkacommon.DefaultDeliveryTimeout)
+		if err != nil {
```
[the remark this answers is not in the document]
Comment (ldjebran) [reply pkg/services/images.go:1630]: if there is no delivery error return message.TopicPartition.Error will be nil , that value is returned here 
https://github.com/RedHatInsights/edge-api/pull/1878/files#diff-a729e21b3416ffbb0d06fbea95ca628e78bbccc429ed70626814d6d09dfbb653R30

[the remark this answers is not in the document]
Comment (AaronH88) [reply pkg/services/images.go:1630]: got it, thanks!

**Review by AaronH88 — DISMISSED** [review]
/lgtm

**Review by AaronH88 — APPROVED** [review]
/lgtm

File: pkg/services/images.go [hunk 3]
```diff
@@ -1610,17 +1610,29 @@ func (s *ImageService) SendImageNotification(i *models.Image) (ImageNotification
...
+		}, deliveryChan)
 
 		if perr != nil {
-			s.log.WithField("message", perr.Error()).Error("Error on produce")
+			s.log.WithFields(log.Fields{"message": perr.Error(), "topic": topic}).Error("Error on produce")
+			return notify, perr
+		}
+
+		// wait for notification delivery
+		err = WaitNotificationDelivery(deliveryChan, kafkacommon.DefaultDeliveryTimeout)
```
Comment (mgold1234) [inline pkg/services/images.go:1630]: can you please add test for that, as you did in the update?

Comment (ldjebran) [reply pkg/services/images.go:1630]: There is no test for this function as use the old producer and not using our new approach, do not want to write a new unittest as this function will be refactored to use new approach, but we want to understand the notification are  delivered or not. We do not want to refactor this code at this stage.

Comment (ldjebran) [reply pkg/services/images.go:1630]: if this is a blocker. I will remove the waitDelivery from images.

**Review by mgold1234 — APPROVED** [review]
/lgtm

**Pull request merged**